addons/survey/models/survey_survey.py

```diff
@@ -426,14 +426,19 @@ def prepare_result(self, question, current_filters=None):
     @api.model
     def get_input_summary(self, question, current_filters=None):
         """ Returns overall summary of question e.g. answered, skipped, total_inputs on basis of filter """
-        current_filters = current_filters if current_filters else []
-        result = {}
-        search_line_ids = current_filters if current_filters else question.user_input_line_ids.ids
-
-        result['answered'] = len([line for line in question.user_input_line_ids if line.user_input_id.state != 'new' and not line.user_input_id.test_entry and not line.skipped])
-        result['skipped'] = len([line for line in question.user_input_line_ids if line.user_input_id.state != 'new' and not line.user_input_id.test_entry and line.skipped])
+        domain = [
+            ('user_input_id.test_entry', '=', False),
+            ('user_input_id.state', '!=', 'new'),
+            ('question_id', '=', question.id)
+        ]
+        if current_filters:
+            domain = expression.AND([[('id', 'in', current_filters)], domain])
 
-        return result
+        line_ids = self.env["survey.user_input_line"].search(domain)
+        return {
+            'answered': len(set([l.user_input_id for l in line_ids if not l.skipped])),
+            'skipped': len(set([l.user_input_id for l in line_ids if l.skipped])),
```

Comment:
Actually, since we're filtering recordset, shouldn't this be:

```py
return {
    'answered': len(line_ids.filtered(lambda line: not line.skipped).mapped('user_input_id')),
    'skipped': len(line_ids.filtered(lambda line: line.skipped).mapped('user_input_id'))
}
```
Building a set of a list of a recordset feels strange, since recordsets are basically sets already.

Side note: I'm pretty sure "mapped" should remove duplicate user_input records, but you have a test case to verify this anyway.

Reply:
Ah yes, it's far better, thanks :+1: